### DISCUSSION TITLE
Recommendations on clocks for multiple icecast outputs

### DISCUSSION THREAD

**Discussion by KlaasT** [savonet/liquidsoap#2987] [User Support]
Hi,

I've currently got a dynamically generated liquidsoap script which outputs a stream towards multiple Icecast servers. All wit the same bit and sample rate.

What I am fearing is if one of the servers starts lagging all are suffering as all are sharing the same clock? Please correct me if I'm wrong.

What are the general recommendations on this kind of setup (multiple icecast outputs). From what I understood from various Stackoverflow and script samples it is better to give each output their own clock.

But is this information correct?

Thanks very much in advance.

**Comment by toots** [comment]
Hi @KlaasT ! Sorry for the late response.

You assumption is correct. I would put each icecast output in their own clock. I believe the most simple solution is to wrap their source in a `buffer`:

```liquidsoap
output.icecast(
   ...,
  buffer(s)
)
```

**Reply by KlaasT** [reply]
No worries. Thanks for your advice though :)

I indeed now added a clock(buffer(10.,s)) to my outputs and it removed many issues with "we need to catch up...".

In your sample you didn't add a buffer time. Is there any particular reason for it?

**Reply by toots** [reply]
These two should be equivalent. `buffer(s)` get their own clock implicatively otherwise. With `clock` you can specify a clock name which can be useful for debugging/logging.
